### PR TITLE
E2E Test: YAML editor

### PR DESCRIPTION
**Pull request by torchiaf**
### Summary
Related issue: https://github.com/rancher/dashboard/issues/8999

### Occurred changes and/or fixed issues

It adds e2e tests for `ResourceYaml` component.

### Technical notes summary

- Check body and footer to be visible at human eye.
- Add new `components` tag for UI elements.
- Add new Cypress Command to check elements visibility.

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Comment by richard-cox** [comment]
@torchiaf Good spot, just let me know when you need it done

File: cypress/support/commands/chainable.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+/**
+ * Check if an element is visible to the user on the screen.
+ */
+Cypress.Commands.add('isVisible', { prevSubject: true }, (subject) => {
+  const bottom = Cypress.$(cy.state('window')).height();
+  const rect = subject[0].getBoundingClientRect();
+
+  expect(rect.top).not.to.be.greaterThan(bottom);
+  expect(rect.bottom).not.to.be.greaterThan(bottom);
+  expect(rect.top).not.to.be.greaterThan(bottom);
```
Comment (aalves08) [inline cypress/support/commands/chainable.ts:11]: aren't these duplicated or am I misreading this?

**Review by aalves08 — COMMENTED** [review]
Left a comment @torchiaf

File: cypress/support/commands/chainable.ts (hunk 1 above)
Comment (torchiaf) [reply cypress/support/commands/chainable.ts]: Thanks, it's a typo, I've removed the duplicate rows.

**Review by aalves08 — APPROVED** [review]
LGTM

**Comment by torchiaf** [comment]
> @torchiaf Good spot, just let me know when you need it done

@richard-cox now that is merged, we should set new `@component` tag as mandatory to unblock the pull requests merge.

**Pull request merged**